src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -113,17 +113,22 @@ public boolean doSubscriptionExist(String subscriptionName) {
     public boolean modifySubscription(Subscription subscription, String subscriptionName) {
         ObjectMapper mapper = new ObjectMapper();
         Document result = null;
+        String query;
         try {
             String stringSubscription = mapper.writeValueAsString(subscription);
             String ldapUserName = (ldapEnabled) ? HttpSessionConfig.getCurrentUser() : "";
-            String query = generateQuery(subscriptionName, ldapUserName);
+            boolean ownerExist = doSubscriptionOwnerExist(subscriptionName);
+            if (!ownerExist) {
```

Comment:
This if-statement is duplicated, please move the if-statement to be inside the generateQuery function, cause in that case then it only needs to be added once.

Reply:
I have created a new method so that IF is used only once